### PR TITLE
feat: torch segmentation model prediction

### PR DESCRIPTION
**Pull request by beniz**
This PR adds support for prediction with DD for torchvision's segmentation models.

Remaining features:
- [x] unit tests
- [x] confidence map / per pixel
- [x] list of torchvision segmentation models with backbone in export script
- [x] have caffe segmentation `img_resize` use the global `img_resize_vector` function.

Example:
- Model tracing:
```
python3 trace_torchvision.py deeplabv3_resnet50 --num_classes 21
```

- Model inference:
```
python3 segment.py --nclasses 21 --model-dir /path/to/model/ --image /path/to/cat.1960.jpg --mllib torch --rgb --width 224 --height 224 --scale 0.0044
```
and for the confidence map:
```
python3 segment.py --nclasses 21 --model-dir /path/to/model/ --image /path/to/cat.1960.jpg --mllib torch --rgb --width 224 --height 224 --scale 0.0044 --confidences best
```

- Results:
image

### CODE REVIEW

**Comment by beniz** [comment]
@mergifyio update

File: demo/segmentation/segment.py [hunk 1]
```diff
@@ -47,13 +54,15 @@ def random_color():
     pass
 
 # prediction call
-parameters_input = {'segmentation':True}
-parameters_mllib = {'gpu':True,'gpuid':0}
-parameters_output = {}
+parameters_input = {'segmentation':True,'scale':args.scale,'rgb':args.rgb}
+parameters_mllib = {'segmentation':True}
```
Comment (Bycob) [inline demo/segmentation/segment.py:58]: Do we really need to have `segmentation` twice here?

File: src/backends/torch/torchlib.cc [hunk 2]
```diff
@@ -1511,6 +1525,119 @@ namespace dd
...
+                /*std::cerr << "segmentation output dict size="
+                          << out_dict.size() << std::endl;
+                auto od_it = out_dict.begin();
+                while (od_it != out_dict.end())
+                  {
+                    std::cerr << (*od_it).key() << std::endl;
+                    ++od_it;
+                  }
+                  std::cerr << std::endl;*/
+                // debug
```
Comment (Bycob) [inline src/backends/torch/torchlib.cc:1542]: Remove debug stuff before merge?

File: tools/torch/trace_torchvision.py [hunk 3]
```diff
@@ -255,8 +265,9 @@ def get_detection_input():
 
         # TODO try scripting instead of tracing
         example = torch.rand(1, 3, 224, 224)
-        script_module = torch.jit.trace(model, example)
-
+        #script_module = torch.jit.trace(model, example)
```
Comment (Bycob) [inline tools/torch/trace_torchvision.py:268]: Remove tracing completely? Or add as an option (but we can do that later if we need it for a specific model)

File: demo/segmentation/segment.py (hunk 1 above)
Comment (beniz) [reply demo/segmentation/segment.py:58]: No we don't, will fix.

File: tools/torch/trace_torchvision.py (hunk 3 above)
Comment (beniz) [reply tools/torch/trace_torchvision.py:268]: Done.

**Review by Bycob — APPROVED** [review]

**Comment by beniz** [comment]
@mergifyio update

**Pull request merged**